### PR TITLE
refactor: register virtual machine

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Description

- Moves the destroy timeout value to a constant.
- Avoids logging an error before checking if it exists.
- Combines the timeout error message logging and UI notification: Since both actions occur together, keep them within the same conditional block to avoid repetition.
- Uses a ticker for periodic checks instead of manually sleeping and checking. This is more idiomatic and readable.

### Testing

Standard

```console
packer-plugin-vmware on  refactor/register
➜ go fmt ./...                                 

packer-plugin-vmware on  refactor/register
➜ make generate
2024/07/08 14:10:42 Copying "docs" to ".docs/"
2024/07/08 14:10:42 Replacing @include '...' calls in .docs/
Compiling MDX docs in '.docs' to Markdown in '.web-docs'...

packer-plugin-vmware on  refactor/register
➜ make build

packer-plugin-vmware on  refactor/register
➜ make test
?       github.com/hashicorp/packer-plugin-vmware       [no test files]
?       github.com/hashicorp/packer-plugin-vmware/version       [no test files]
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/common 7.021s
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/iso    2.631s
ok      github.com/hashicorp/packer-plugin-vmware/builder/vmware/vmx    3.403s

packer-plugin-vmware on  refactor/register 
➜ make dev
packer plugins install --path packer-plugin-vmware "github.com/hashicorp/vmware"
Successfully installed plugin github.com/hashicorp/vmware from /Users/johnsonryan/Library/Mobile Documents/com~apple~CloudDocs/Code/Work/packer-plugin-vmware1/packer-plugin-vmware to /Users/johnsonryan/.packer.d/plugins/github.com/hashicorp/vmware/packer-plugin-vmware_v1.0.12-dev_x5.0_darwin_amd64
```

### CODE REVIEW

File: builder/vmware/common/step_register.go [hunk 1]
```diff
@@ -52,35 +54,45 @@ func (s *StepRegister) Cleanup(state multistep.StateBag) {
...
 
 			s.registeredPath = ""
 		} else {
-			ui.Say("Destroying virtual machine...")
+			ui.Say("Deleting virtual machine...")
 			if err := remoteDriver.Destroy(); err != nil {
-				ui.Error(fmt.Sprintf("Error destroying VM: %s", err))
+				errorMessage := fmt.Sprintf("error deleting virtual machine: %s", err)
+				ui.Error(errorMessage)
+				log.Printf(errorMessage)
```
Comment (lbajolet-hashicorp) [inline builder/vmware/common/step_register.go:76]: I would remove the `log.Printf` if there's already a `ui.Error`, as they'll be visible in any case

**Review by lbajolet-hashicorp — APPROVED** [review]
Just left one comment regarding usage of `log.Printf` in addition to `ui.Error` as it feels like a suspender+belt kind of situation, but aside from that LGTM!
Pre-approving to not delay a later merge.

File: builder/vmware/common/step_register.go (hunk 1 above)
Comment (tenthirtyam) [reply builder/vmware/common/step_register.go]: Resolved in 13c8c54 .

**Comment by lbajolet-hashicorp** [comment]
FYI: I replaced a couple of `ui.Error(fmt.Sprintf...` with `ui.Errorf` in a quick last reroll, if tests are green, I'll merge this

**Pull request merged**